src/shapes/cylinder.cpp

```diff
@@ -82,118 +82,127 @@ class Cylinder final : public Shape<Float, Spectrum> {
     using typename Base::ScalarSize;
 
     Cylinder(const Properties &props) : Base(props) {
-        m_radius = props.float_("radius", 1.f);
+        bool has_to_world = props.has_property("to_world");
 
-        ScalarPoint3f p0 = props.point3f("p0", ScalarPoint3f(0.f, 0.f, 0.f)),
-                      p1 = props.point3f("p1", ScalarPoint3f(0.f, 0.f, 1.f));
+        if (!has_to_world) {
```

Comment:
Could we have both the `to_world` transform and the `p0`, `p1` properties at the same time?
In which case we should first build a pre_transform with the `p0` `p1` properties and later multiply the `to_world` transform. See how it is done in `sphere.cpp`.